src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -133,7 +133,11 @@ public virtual void ValidateId(DocumentsOperationContext context, Slice lowerId,
                 ValidateId(context, lowerId, newFlags);
 
                 var collectionName = _documentsStorage.ExtractCollectionName(context, document);
-                var table = context.Transaction.InnerTransaction.OpenTable(_documentDatabase.GetDocsSchemaForCollection(collectionName), collectionName.GetTableName(CollectionTableType.Documents));
+                
+                var table = context.Transaction.InnerTransaction.OpenTable(
+                    (newFlags & DocumentFlags.Archived) == DocumentFlags.Archived
+                        ? _documentDatabase.DocumentsStorage.CompressedDocsSchema
+                        : _documentDatabase.GetDocsSchemaForCollection(collectionName), collectionName.GetTableName(CollectionTableType.Documents));
```

Comment:
create new `GetDocsSchemaForCollection(collectionName, flags)` and handle it there

Reply:
done